src/hooks/useArticleLinksNavigation.js

```diff
@@ -1,18 +1,30 @@
 import { useState, useEffect } from 'preact/hooks'
-import { useSoftkey } from 'hooks'
+import { useSoftkey, usePopup, useI18n } from 'hooks'
 import { viewport } from 'utils'
+import { ArticlePreview } from 'components'
 
 const SELECTED_ATTRIBUTE = 'data-selected'
 
+const SUPPORTED_LINKS = [
+  '[data-action]',
+  'a[title]',
+  '.mw-ref',
+  'a[rel="mw:ExtLink"]'
+].join(',')
+
 export const useArticleLinksNavigation = (
+  origin,
+  lang,
   contentRef,
   currentPage,
-  onTitleClick,
-  onActionClick
+  linkHandlers = {}
 ) => {
+  const i18n = useI18n()
   const [links, setLinks] = useState([])
   const [currentLink, setCurrentLinkInternal] = useState(null)
 
+  const [showArticlePreview] = usePopup(ArticlePreview, { position: 'bottom' })
```

Comment:
sometimes the preview only have few lines, then we will have a lot of blank content. do you think 'auto' here better? 

but then the 'loading' effect feel bad =( (or we can do the same as [loading article](https://phabricator.wikimedia.org/T240888), less than 500ms don't show loading)
![Screen Shot 2019-12-21 at 12 14 33 AM](https://user-images.githubusercontent.com/2560096/71298150-f7e82f00-2386-11ea-81eb-56ff655f2b7a.png)

Reply:
Yeah, it turns out 'auto' is the only mode that makes sense. I'll simplify this.

Reply:
In our target audience, loading is unlikely to be fast. Whatever we do for the main article loading screen, we should do something similar here.